src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let joined = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (i !== this.length - 1) {
+        joined += `${this[i]}${separator}`;
+      } else {
+        joined += `${this[i]}`;
+      }
```

Comment:
and then only add separator here, like:
```suggestion
      if (i !== this.length - 1) {
        joined += separator;
      }
```